velox/exec/tests/PrefixSortTest.cpp

```diff
@@ -0,0 +1,329 @@
+  const RowVectorPtr generateExpectedResult(
+      const std::vector<CompareFlags> compareFlags,
+      const int numRows,
+      const RowVectorPtr& sortedRows);
+
+  // 20 MB
+  const int64_t maxBytes_ = 20LL << 20;
+  const std::shared_ptr<memory::MemoryPool> rootPool_{
+      memory::MemoryManager::getInstance()->addRootPool(
+          "PrefixSortTest",
+          maxBytes_)};
+
+  const std::shared_ptr<memory::MemoryPool> pool_{
+      rootPool_->addLeafChild("PrefixSortTest", maxBytes_)};
+
+  const std::shared_ptr<memory::MemoryPool> fuzzerPool =
+      rootPool_->addLeafChild("VectorFuzzer");
+};
+
+void PrefixSortTest::SetUp() {
+  filesystems::registerLocalFileSystem();
+  if (!isRegisteredVectorSerde()) {
+    this->registerVectorSerde();
+  }
+}
+
+void PrefixSortTest::storeRows(
+    const int numRows,
+    const RowVectorPtr& sortedRows,
+    std::vector<char*>& rows,
+    RowContainer* data) {
+  SelectivityVector allRows(numRows);
+  rows.resize(numRows);
+  for (int row = 0; row < numRows; ++row) {
+    rows[row] = data->newRow();
+  }
+  for (int column = 0; column < sortedRows->childrenSize(); ++column) {
+    DecodedVector decoded(*sortedRows->childAt(column), allRows);
+    for (int i = 0; i < numRows; ++i) {
+      char* row = rows[i];
+      data->store(decoded, i, row, column);
+    }
+  }
+}
+
+const RowVectorPtr PrefixSortTest::generateExpectedResult(
+    const std::vector<CompareFlags> compareFlags,
+    const int numRows,
+    const RowVectorPtr& sortedRows) {
+  const auto sortedRowsType = asRowType(sortedRows->type());
+  const int sortKeysNum = sortedRows->childrenSize();
+  RowContainer rowContainer(sortedRowsType->children(), pool_.get());
+
+  std::vector<char*> rows;
+  storeRows(numRows, sortedRows, rows, &rowContainer);
+  std::sort(
+      rows.begin(), rows.end(), [&](const char* leftRow, const char* rightRow) {
+        for (vector_size_t index = 0; index < sortKeysNum; ++index) {
+          if (auto result = rowContainer.compare(
+                  leftRow, rightRow, index, compareFlags[index])) {
+            return result < 0;
+          }
+        }
+        return false;
+      });
+  const RowVectorPtr result =
+      BaseVector::create<RowVector>(sortedRowsType, numRows, pool_.get());
+  for (int column = 0; column < sortedRows->childrenSize(); ++column) {
+    rowContainer.extractColumn(
+        rows.data(), numRows, column, result->childAt(column));
+  }
+  return result;
+}
+
+void PrefixSortTest::testSort(
+    const int numRows,
+    const std::vector<TestSetting> testSettings,
+    const RowVectorPtr& sortedRows) {
+  const auto sortedRowsType = asRowType(sortedRows->type());
+  for (const auto& setting : testSettings) {
+    SCOPED_TRACE(setting.debugString());
+    // 1. Store test rows into rowContainer.
+    RowContainer rowContainer(sortedRowsType->children(), pool_.get());
+    std::vector<char*> rows;
+    storeRows(numRows, sortedRows, rows, &rowContainer);
```

Comment:
Perhaps, modify storeRows to return std::vector<char*> instead of taking it as output parameter.